frmts/heif/heifdataset.cpp

```diff
@@ -700,5 +654,9 @@ void GDALRegister_HEIF()
 
     poDriver->pfnOpen = GDALHEIFDataset::Open;
 
+#ifdef HAS_CUSTOM_FILE_WRITER
```

Comment:
HEIFDriverSetCommonMetadata() should set GDAL_DCAP_CREATECOPY when HAS_CUSTOM_FILE_WRITER i set,  for deferred loaded plugins (https://gdal.org/development/rfc/rfc96_deferred_plugin_loading.html)

Reply:
Implemented, to be tested.

Reply:
Tested